networkconfig.py

```diff
@@ -3,201 +3,304 @@
-		<h1 style="color: #5e9ca0; text-align: center;"><span style="color: #ff0000;">Wi-Fi Not Configured to """+ssid+"""</span></h1>
-		<br><br>
-		<form>
-			<input type="button" value="Go back!" onclick="history.back()"></input>
-		</form></center></html>
-		"""
-		send_response(client, response_footer )
-		return (False)
+    match = ure.search("ssid=([^&]*)&password=(.*)", request)
+
+    if match is None:
+        send_response(client, "Parameters not found", status_code=400)
+        return False
+    # version 1.9 compatibility
+    try:
+        ssid = match.group(1).decode("utf-8").replace("%3F", "?").replace("%21", "!")
+        password = match.group(2).decode("utf-8").replace("%3F", "?").replace("%21", "!")
+    except:
+        ssid = match.group(1).replace("%3F", "?").replace("%21", "!")
+        password = match.group(2).replace("%3F", "?").replace("%21", "!")
+
+    if len(ssid) == 0:
+        send_response(client, "SSID must be provided", status_code=400)
+        return False
+
+    if do_connect(ssid, password):
+        response = """\
+            <html>
+                <center>
+                    <br><br>
+                    <h1 style="color: #5e9ca0; text-align: center;">
+                        <span style="color: #ff0000;">
+                            ESP successfully connected to WiFi network %(ssid)s.
+                        </span>
+                    </h1>
+                    <br><br>
+                </center>
+            </html>
+        """ % dict(ssid=ssid)
+        send_response(client, response)
+        try:
+            profiles = read_profiles()
+        except OSError:
+            profiles = {}
+        profiles[ssid] = password
+        write_profiles(profiles)
+        return True
+    else:
+        response = """\
+            <html>
+                <center>
+                    <h1 style="color: #5e9ca0; text-align: center;">
+                        <span style="color: #ff0000;">
+                            ESP could not connect to WiFi network %(ssid)s.
+                        </span>
+                    </h1>
+                    <br><br>
+                    <form>
+                        <input type="button" value="Go back!" onclick="history.back()"></input>
+                    </form>
+                </center>
+            </html>
+        """ % dict(ssid=ssid)
+        send_response(client, response)
+        return False
+
 
 def handle_not_found(client, url):
-	send_response(client, "Path not found: {}".format(url), status_code=404)
+    send_response(client, "Path not found: {}".format(url), status_code=404)
+
 
 def stop():
-	global server_socket
+    global server_socket
+
+    if server_socket:
+        server_socket.close()
+        server_socket = None
 
-	if server_socket:
-		server_socket.close()
 
 def start(port=80):
-	addr = socket.getaddrinfo('0.0.0.0', port)[0][-1]
-
-	global server_socket
-	global wlan_sta
-	global wlan_ap
-	global ssid_name
-	global ssid_password
```

Comment:
you only need global if you assign to the name.
if you just use the name, it will use the global anyway if there is no such local name.